projects/client/src/lib/components/buttons/popup/PopupMenu.svelte

```diff
@@ -0,0 +1,168 @@
+<script lang="ts">
+  import MoreIcon from "$lib/components/icons/MoreIcon.svelte";
+  import { clickOutside } from "$lib/utils/actions/clickOutside";
+  import { disableTransitionOn } from "$lib/utils/actions/disableTransitionOn";
+  import { slide } from "svelte/transition";
+  import type { PopupMenuProps } from "./PopupMenuProps";
+  import { usePortalActions } from "./_internal/usePortalActions";
+
+  const { items, onclick, ...props }: PopupMenuProps = $props();
+
+  const { portalHandler, portal, isPopupOpen } = usePortalActions();
+</script>
+
+<div
+  class="trakt-popup-button-container"
+  class:is-popup-open={$isPopupOpen}
+  use:clickOutside
+  onclickoutside={(_) => {
+    isPopupOpen.set(false);
+  }}
+>
+  <button
+    use:disableTransitionOn={"touch"}
+    use:portalHandler
+    aria-haspopup="true"
+    class="trakt-popup-menu-button"
+    onclick={(ev) => {
+      isPopupOpen.update((state) => !state);
+      onclick?.(ev);
+    }}
+    {...props}
+  >
+    <MoreIcon />
+  </button>
+</div>
+
+{#if $isPopupOpen}
+  <div
+    use:portal
+    class="trakt-popup-menu-container"
+    transition:slide={{ duration: 150 }}
+    onoutrostart={(e) => e.currentTarget.classList.add("removing")}
+  >
+    <div class="trakt-popup-menu-button-placeholder"><MoreIcon /></div>
+    <div class="spacer"></div>
+    <ul onclickcapture={() => isPopupOpen.set(false)}>
```

Comment:
this can also be set within the `portal` action so the `isPopupOpen` does not need to be exposed at all.